src/Metadata/GeneratorMetadata.php

```diff
@@ -68,10 +69,14 @@ public function isSourceUserDefined(): bool
 
     public function hasConstructor(): bool
     {
-        if (!$this->allowConstructor) {
+        if ($this->constructorStrategy === ConstructorStrategy::NEVER) {
```

Comment:
in which case would we want to never use the constructor? if the target class has a constructor, wouldn't it break?

Reply:
Depends on what you use your mapping for, by example doctrine never use the constructor when instantating an entity from the database. At least having this, ensure a consistent behavior there is no "magic"